### DISCUSSION TITLE
[提问]如何配置ipv6下的透明代理

### DISCUSSION THREAD

**Discussion by nJhEqnvK** [XTLS/Xray-core#162] [Q&A]
因为最近看到ipv6的相关bug，如 https://github.com/XTLS/Xray-core/issues/137 、 https://github.com/XTLS/Xray-core/issues/48 等已经被修复，就在想配置ipv6试试（因为家里是移动的宽带，ipv4只有私网地址，ipv6能拿到`2409:`的公网，于是就照着ipv4的tproxy写了规则。

目前遇到的问题是，ipv6无法像ipv4那样直接通过DHCP option3来指定默认网关到旁路由（运行Xray）上，还想请教各位有什么建议或解决方案吗？ 

提前感谢，后附我瞎写的透明代理脚本

```bash
ip -6 route add local default dev lo table 100
ip -6 rule add fwmark 1 lookup 100

# 代理局域网设备
ip6tables -t mangle -N XRAY

ip6tables -t mangle -A XRAY -d fd00::/8 -p tcp -j RETURN # 直连局域网，避免 XRAY 无法启动时无法连网关的 SSH，如果你配置的是其他网段（如 10.x.x.x 等），则修改成自己的
ip6tables -t mangle -A XRAY -d fd00::/8 -p udp -j RETURN # 直连局域网，53 端口除外（因为要使用 XRAY 的 

ip6tables -t mangle -A XRAY -m mark --mark 0xff -j RETURN 
ip6tables -t mangle -A XRAY -p udp -j TPROXY --on-port 12345 --tproxy-mark 1 # 给 UDP 打标记 1，转发至 12345 端口
ip6tables -t mangle -A XRAY -p tcp -j TPROXY --on-port 12345 --tproxy-mark 1 # 给 TCP 打标记 1，转发至 12345 端口
ip6tables -t mangle -A PREROUTING -j XRAY # 应用规则
```

**Comment by badO1a5A90** [comment]
我觉得不是发在哪个区的原因，是这道题实际有操作经验的人太少了。

**Comment by elvisw** [comment]
参考这个项目：
https://github.com/zfl9/ss-tproxy

**Comment by EkkoG** [comment]
https://github.com/xiaorouji/openwrt-passwall 目前在用这个项目，整体来说，体验接近完美了，楼主可以参考一相相关的配置，本人没看太懂，想把这个项目的配置搬到 clash 那边没有成功

**Comment by BioniCosmos** [comment]
我来挖个坟。

旁路由上使用 IPv6 透明代理不可能像 IPv4 一样仅设置默认网关就能成功，具体原理我也不好说。但有一个折衷方法：使用 4 over 6 隧道。这个方法不需要有公网 IPv6 的支持，具体来说即使有也用不上。思路是在局域网内使用 ULA 分配 IPv6 地址，可以使用 radvd 进行分配，也可以手动配置，然后通过 *ray 通过 IPv4 传输 IPv6 到服务器，如果服务器有原生 IPv6 最好，没有的话可以用 CF Warp。

此方法已测试成功，但目前有一些小问题，详情可以询问我。

**Reply by chise0713** [reply]
看看实操

**Reply by BioniCosmos** [reply]
竟然还有人挖。

**Comment by BioniCosmos** [comment]
未经过测试，可能有用的解决措施：[在不关闭主路由 SLAAC 的前提下对家庭内网指定设备的IPv6流量进行“旁路牵引”](https://blog.ihipop.com/2022/09/5343.html)
